### PR TITLE
Fix #13491: Add regression test

### LINKED ISSUE #13491: 3.0.2 typing incompatible with 3.0.1
## Compiler version

Our ongoing effort to port parboiled2 (and maybe more importantly akka-http and depending projects) to Scala 3, is hindered by the fact that code that compiled with 3.0.1, does not type correctly any more with 3.0.2.

Admittedly, the typing code is sophisticated, so we didn't expect that it would work unchanged from Scala 2 before. We rewrote some of the heavy implicit machinery using type matches which seemed to work somewhat with 3.0.1 (aside from some typing errors, stack overflows in the typer, and typing termination problems that we ignored or worked around for now).

## Reproducer

See https://github.com/sirthias/parboiled2/pull/280, and use Scala 3.0.2 instead.

## Output

Lots of type errors.

## Expectation

Typing correctly.

### PR DESCRIPTION
**Pull request by OlivierBlanvillain**

### CODE REVIEW

**Review by smarter — APPROVED** [review]
LGTM although it might be useful to identify which PR fixed this to backport it to 3.1.0 since it fixes a regression.

File: compiler/test/dotc/pos-test-pickling.blacklist [hunk 1]
```diff
@@ -37,6 +37,7 @@ i9999.scala
 12093.scala
 9757.scala
 9890.scala
+13491.scala
```
Comment (dwijnand) [inline compiler/test/dotc/pos-test-pickling.blacklist:40]: How come?

**Comment by OlivierBlanvillain** [comment]
Fix bisect to https://github.com/lampepfl/dotty/pull/12773

File: compiler/test/dotc/pos-test-pickling.blacklist (hunk 1 above)
Comment (OlivierBlanvillain) [reply compiler/test/dotc/pos-test-pickling.blacklist:40]: The pickling test failure can be minimized to the following: 

```scala
object Test {
  type M[X] = X match { case Int => String }
  def a: Any { type Out = M[Int] } = a
}
```

The failure comes from the refinement type, which is printed as
`scala.Any{Out = scala.String}` before pickling, and
`scala.Any{Out = Test.M[scala.Int]}` after pickling...

I'll open a new issue from that minimization.

**Pull request merged**